variables.tf

```diff
@@ -1018,6 +1145,26 @@ variable "sticky_settings" {
   DESCRIPTION
 }
 
+variable "storage_account" {
```

Comment:
what's the difference between storage_account, storage_account_name, and storage_accounts?

If you are going to create a storage account then encapsulate all required inputs into an object, then pass to the storage account module.

If you are going to accept an existing storage account then pass by resource id and don't use a data source. Use a validation rule to ensure the resource id is valid, then split the string to get the RG and name

Reply:
@matt-FFFFFF - apologies for the long-winded response:

The `storage_account` variable works with `create_storage_account`, and allows creation/configuration of a new storage account within the module itself. The thought-process here is that you need a storage account for the `functionapp` and the portal allows you to create a new storage account within the deployment pane.

For the `functionapp`, you can specify an existing storage account using `storage_account_name` with either `storage_account_access_key` or `storage_usues_managed_identity` (based on the azurerm documentation for `azurerm_[ linux | windows ]_function_app`). So, as you stated, I may need to change the variable for the resource id and extract the name, but then they would still need to supply access key potentially, unless I leverage a data object. I was just trying what I thought would be easiest.

The `storage_accounts` variable refers to the `storage_account` block within the actual azurerm resource (in the azurerm documentation, this would be the `storage_account` block for the`azurerm_[ linux | windows ]_function_app` used for mounting file shares which is only for Linux, but not actually a full storage account). It might be beneficial to rename this on my end, but I wanted to still have the functionality for mounting multiple file shares, as well as creating the one needed storage account for the `functionapp`.

Reply:
I think you could implement this as follows:

rename `storage_account` to `function_app_storage_account`

rename `create_storage_account` to `function_app_create_storage_account`

rename `storage_account_name` to `function_app_storage_account_name`

rename `storage_account_access_key` to `function_app_storage_account_access_key`

rename `storage_usues_managed_identity` to `function_app_storage_uses_managed_identity`

I would not use a data source, they can be very problematic.

As for the `storage_accounts` variable - I feel like this needs a more specific name, or at least some good description text in the variable declaration.